### PR TITLE
DCES-354 Positive DRC logging integration test

### PR DESCRIPTION
**Pull request by dconneely**
## What

[DCES-354](https://dsdmoj.atlassian.net/browse/DCES-354) Positive DRC logging response integration test

- Added a new integration test and corresponding spy for the DRC logging endpoint
- Moved utility method to updateStatus into SpyFactory utility class
- Changed `WebClientException` to `WebClientResponseException` in services

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

[DCES-354]: https://dsdmoj.atlassian.net/browse/DCES-354?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/service/DrcLoggingIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                            .content(json))
+                    .andExpect(status().isOk())
+                    .andExpect(content().string("The request has been processed successfully"));
+        } catch (Exception e) {
+            softly.fail("acknowledgeContributionMVC(" + concorContributionId + ") failed", e);
+        }
+    }
+
+    /*
+    // Need to handle CSRF and OAuth2 login before can use testRestTemplate like this:
```
Comment (LAA-Dave-Edwards) [inline dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/service/DrcLoggingIntegrationTest.java:155]: Good to remove the commented out code? Or is this valuable as assistance?

Comment (dconneely) [reply dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/service/DrcLoggingIntegrationTest.java:155]: Silly me. I thought I'd removed it. Will fix this and the type decl on other PR on Monday am

File: dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/testing/DrcLoggingProcessSpy.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        private DrcLoggingProcessSpyBuilder() {
+            throw new UnsupportedOperationException("Call SpyFactory.newDrcLoggingProcessSpyBuilder instead");
+        }
+
+        DrcLoggingProcessSpyBuilder(final ContributionClient contributionClientSpy) {
+            this.contributionClientSpy = contributionClientSpy;
+        }
+
+        public void instrumentSendLogContributionProcessed() {
```
Comment (LAA-Dave-Edwards) [inline dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/testing/DrcLoggingProcessSpy.java:49]: Again, use of instrument?

**Review by LAA-Dave-Edwards — CHANGES_REQUESTED** [review]
Looks good, just some clarifications.

File: dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/testing/ContributionProcessSpy.java [hunk 3]
```diff
@@ -59,7 +63,18 @@ public void instrumentGetContributionsActive() {
             }).when(contributionClientSpy).getContributions("ACTIVE");
         }
 
-        public void instrumentStubbedSendContributionUpdate(Boolean stubbedResult) {
+        public void instrumentAndFilterGetContributionsActive(final List<Integer> activeIds) {
```
[the remark this answers is not in the document]
Comment (dconneely) [reply dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/testing/ContributionProcessSpy.java:66]: `instrument` meaning monitor and keep track of the arguments and return values.

File: dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/testing/DrcLoggingProcessSpy.java (hunk 2 above)
Comment (dconneely) [reply dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/testing/DrcLoggingProcessSpy.java:49]: In this case, it again means keeping track of the args into, and return-value from, that method (so the test can examine them later). I guess could use the word `track` instead? Let me know what you think.

**Comment by dconneely** [comment]
@LAA-Dave-Edwards Not sure why I could merge #48 without getting a re-review, but I can't merge this one (#49) without it.

**Review by LAA-Dave-Edwards — APPROVED** [review]

**Pull request merged**